group_project/group_activity.py

```diff
@@ -1,3 +1,75 @@
+"""
+This module parses Group project XML into a tree of domain-specific objects.
+Basically it looks like the following
+
+GroupActivity (.)
+- ActivityComponent (./projectcomponent)
+-- ActivitySection (//section - different paths)
+--- ActivityQuestion (//section/question)
+---- <no dedicated class, etree.Element is used> (//section/question/answer)
+--- ActivityAssessment (//section/assessment)
+---- <no dedicated class, etree.Element is used> (//section/assessment/answer)
+
+GroupActivity (paths relative to root element)
+- resources: [DottableDict(title, description, location)] - ./resources/document
+- grading_criteria: [DottableDict(title, description, location)] - ./resources/document[@grading_criteria="true"]     # attribute filter implicit
+- milestone_dates: {str: datetime.date} - ./dates/milestone
+- submissions: [DottableDict(id, title, description, location?)] - ./submissions/document
+- grade_questions: [ActivityQuestion] - //section/question
+- activity_components: [ActivityComponent] - ./projectcomponent
+- grading_override: Bool                                                                # if True - allows visiting components after close date; used by TA
+* has_submissions: Bool                                                                 # True if ANY submission uploaded
+* has_all_submissions: Bool                                                             # True if ALL submission uploaded
+* submission_json: json                                                                 # submissions serialized into json format
+* step_map: json
+    {
+        component_id: { prev: prev_component.id, name: component.name, next: next_component.id},
+        ordered_list: [component1.id, component2.id, ...],
+        default: <latest_open_component.id if not grading_override else latest_component_with_other_group_sections.id>
+    }
+
+ActivityComponent (paths relative to ./projectcomponent)
+-- id: str - ./@id
+-- name: str - ./@name
+-- sections: [ActivitySection] - ./section
+-- peer_review_sections: [ActivitySection] - ./peerreview/section                       # peer review - providing feedaback to teammates
+-- peer_assessment_sections: [ActivitySection] - ./peerassessment/section               # peer review - reviewing teammates' feedaback on curernt student
+-- other_group_sections: [ActivitySection] - ./projectreview/section                    # group review - providing feedaback to other groups
+-- other_group_assessment_sections: [ActivitySection] - ./projectassessment/section     # group review - reviewing other groups' feedaback on curernt group
```

Comment:
If we follow the recommendation to have different stages types, the presence of these fields would depend on the stage type - each stage type would allow different types of children, depending on whether it's a peer assessment, evaluation, etc.